### ISSUE TITLE
Does Readwise Mirror support adding links to "View Highlight in Reader"?

### ISSUE THREAD

**Issue by tdnzr** [jsonMartin/readwise-mirror#29]
For highlights made in Readwise Reader, the official Readwise-Obsidian plugin appends a link to the Reader highlight, like so: `([View Highlight](https://read.readwise.io/read/some_string))`

I find this functionality very useful, and would like to implement it in Readwise Mirror. Is this a) not possible in the plugin (e.g. because Readwise doesn’t expose the corresponding API), b) conceptually possible but not implemented in the plugin, or c) actually already possible?

**Comment by jsonMartin** [comment]
Hi @tdnzr , I just investigated this for you and it appears that it's looking like answer A.

As you guessed, it appears that this particular field is not exposed by the APIs used in this plugin. It looks like the field you're looking for is `readwise_url`; however, that field is only exposed on the `Export` API which sends the entire Readwise library (which is how other plugins, such as Notion's work).

However, this plugin uses Readwise's "Advanced" API in order to only sync updated data, and that includes different fields. This is what `highlights` API returns:
```
{
  "id": 13,
  "text": "To be alone for any length of time is to shed an outer skin. The...",
  "note": "",
  "location": 57,
  "location_type": "location",
  "highlighted_at": "2020-02-02T16:46:07Z",
  "url": null,
  "color": "yellow",
  "updated": "2020-04-06T12:30:52.318552Z",
  "book_id": 1337,
  "tags": []
}
```

You can check out their API documentation here: https://readwise.io/api_deets

**Comment by tdnzr** [comment]
1) Thanks for the links to the Readwise API documentation. I think you're right that the Advanced API doesn't provide what I want, but I don't think the `readwise_url` field is what I want, because it doesn't actually link to the aforementioned URL `https://read.readwise.io/read/some_string`. The closest guess I have for what I actually want is that `some_string` might be the same as the `external_id` field exposed in the Export API.

2) That said, in your example above, what is the "url" field in the highlights API? It's `null` in your example, and it's not even mentioned in your [readme](https://github.com/jsonMartin/readwise-mirror#highlights), but is it ever not null? Maybe if one makes highlights in Readwise Reader, this might be the Reader URL `https://read.readwise.io/read/some_string`?

3) Finally, the API docs mention that one can use the Export API for syncing updated data, i.e. without fetching the entire library each time, by providing extra parameters:

> Parameters:
> 
> - `updatedAfter` – (Optional, Formatted as ISO 8601 date) Fetch only highlights updated after this date.
> - `ids` – (Optional) Comma-separated list of `user_book_ids`, returns all highlights for these books only.
> - `pageCursor` – (Optional) A string returned by a previous request to this endpoint. Use it to get the next page of books/highlights if there are too many for one request.
> 
> The recommended way to use this endpoint is to first sync all of a user's historical data by passing no parameters on the first request, then pageCursor until there are no pages left. Then later, if you want to pull newly updated highlights, just pass updatedAfter as the time you last pulled data. This is shown in the examples on the right.

**Comment by tdnzr** [comment]
Hey there, I've asked on Discord, and apparently the `url` endpoint is indeed the thing I care about. So this issue can indeed be implemented in Readwise Mirror.

Quoting from Discord [here](https://discord.com/channels/886992134505398314/893104466222542900/1162754386376798440)

My question:
> In the Readwise export API, is there an equivalent to the field "highlight_location_url", which is used by the official Readwise export plugins to display the "View Highlight" (= in Reader) URL (https://read.readwise.io/read/some_string )? Having browsed the Readwise API at https://readwise.io/api_deets, I can't find such an equivalent. My best guess is that possibly, the external_id field might be equivalent to some_string in the aforementioned Reader URL.

Answer:

> It's the highlight_url in the Highlight CREATE endpoint, which is the "[u]nique url of the specific highlight (eg. a concrete tweet or a podcast snippet)".
> If you're looking for the corresponding key in the response, it's "url": in the Highlights LIST endpoint and a few others.

**Comment by tdnzr** [comment]
As you can see above, I've tried to make pull requests based on the info from Discord. Once again apologies for making single-file pull requests.

**Comment by tdnzr** [comment]
For documentation, this issue was implemented in this commit: [943f91a](https://github.com/jsonMartin/readwise-mirror/commit/943f91ac094e4c36ed261eaba1cfd7784a1f16dc)
